### PR TITLE
[2.0.0] Add default max concurrency parameter value

### PR DESCRIPTION
**Pull request by emartynov**
## Task
Implementing `flatMap` changes from #620 

## Changes
* Add default parameter value for `maxConcurrency`
* Update test to compile

## Notes
* This change required test code changes. This is probably fine since it is 2.0 version
* The IDE showed me the warning about unused functions. Should we use `@Suppress` for this issue in these files?

### CODE REVIEW

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt [hunk 1]
```diff
@@ -3,21 +3,14 @@ package com.badoo.reaktive.observable
...
- * Please refer to the corresponding RxJava [document](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#flatMapCompletable-io.reactivex.functions.Function-).
- */
-fun <T> Observable<T>.flatMapCompletable(mapper: (T) -> Completable): Completable =
-    flatMapCompletable(maxConcurrency = Int.MAX_VALUE, mapper = mapper)
-
 /**
  * Calls the [mapper] for each element emitted by the [Observable] and subscribes to the returned inner [Completable].
  * The maximum number of concurrently subscribed inner [Completable]s is determined by the [maxConcurrency] argument.
  *
+ * All inner [Completable]s are subscribed concurrently without any limits.
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt:10]: Add "By default, " here and for other variants?

**Review by arkivanov — CHANGES_REQUESTED** [review]

**Comment by arkivanov** [comment]
> This change required test code changes. This is probably fine since it is 2.0 version

It's fine.

> The IDE showed me the warning about unused functions. Should we use @Suppress for this issue in these files?
No need to suppress, it's ok.

**Comment by arkivanov** [comment]
Also could you please run `apiDump` Gradle task and commit the updated api files?

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt (hunk 1 above)
Comment (emartynov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt:10]: Done

**Comment by emartynov** [comment]
@arkivanov All comments are resolved. Whenever you have time please check.

**Comment by arkivanov** [comment]
@emartynov There are issues found by detekt, looks like some lines are too long now. Could you please put every function argument on new line where the length is exceeded?

> 	MaxLineLength - [flatMap] at /home/runner/work/Reaktive/Reaktive/reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMap.kt:37:1
	MaxLineLength - [flatMapMaybe] at /home/runner/work/Reaktive/Reaktive/reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapMaybe.kt:19:1
	MaxLineLength - [flatMapSingle] at /home/runner/work/Reaktive/Reaktive/reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapSingle.kt:19:1

**Comment by arkivanov** [comment]
Also there are conflicts with the base breach, would it be possible for you to resolve them?

**Comment by emartynov** [comment]
@arkivanov actually, this is ready for review now

**Comment by arkivanov** [comment]
@emartynov There are compilation errors, do you have any spare time to fix?

```
> Task :reaktive:compileDebugKotlinAndroid
e: /home/runner/work/Reaktive/Reaktive/reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt: (15, 71): No type arguments expected for fun Completable.asObservable(): Observable<Nothing>
```

**Comment by emartynov** [comment]
> @emartynov There are compilation errors, do you have any spare time to fix?
> 
> ```
> > Task :reaktive:compileDebugKotlinAndroid
> e: /home/runner/work/Reaktive/Reaktive/reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/FlatMapCompletable.kt: (15, 71): No type arguments expected for fun Completable.asObservable(): Observable<Nothing>
> ```

Checking!

**Comment by emartynov** [comment]
@arkivanov, tests are compiling on my machine. Sorry for such stupid mistake.

**Comment by arkivanov** [comment]
I have restarted the build, let's see if it passes

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for you contribution!

**Pull request merged**